.changeset/little-ads-return.md

```diff
@@ -0,0 +1,73 @@
+---
+"@makeswift/runtime": minor
+---
+
+BREAKING: The latest upgrade to the Makeswift client `getPages` method
+introduces sorting, path filtering, and pagination. This method was not
+previously paginated - in order to get all your pages, you may now use our
+`toArray` pagination helper method, which will automatically paginate through
+all results and aggregate them into an array:
+
+```tsx
+import { client } from '@/makeswift/client'
+import { MakeswiftPageResult } from '@makeswift/runtime/next'
+
+async function getAllPages(): Promise<MakeswiftPageResult[]> {
+  return await client.getPages().toArray()
+}
+```
+
+This change also deprecates the client `getSitemap` method, with the
+recommendation that sitemaps should now be generated using data returned from
+`getPages`. Note that the deprecation of `getSitemap` now involves the host
+being responsible for the construction of page paths in the sitemap (either with
+domain or path based localization). Here's an example that uses path-based
+localization:
+
+See the example below:
+
+```tsx pages/sitemap.xml.tsx
+import { client } from '@/makeswift/client'
+import { MakeswiftPageResult } from '@makeswift/runtime/next'
+import { getServerSideSitemapLegacy } from 'next-sitemap'
+import {getDomainFromLocale} from './utils'
+import "makeswift/components"
+
+const DOMAIN = 'https://example.com'
+const DEFAULT_PRIORITY = 0.75
+const DEFAULT_FREQUENCY = 'hourly'
+
+export async function getServerSideProps(context) {
+  const pages = await client.getPages().toArray()
+
+  return pages
+  .filter(page => !page.excludedFromSearch)
+  .map(page => {
+    const pageUrl = new URL(page.path, DOMAIN)
+    return {
+      loc: pageUrl.href,
+      lastmod: page.createdAt,
+      changefreq: page.sitemapFrequency ?? DEFAULT_FREQUENCY,
+      priority: page.sitemapPriority ?? DEFAULT_PRIORITY,
+      alternateRefs?: page.localizedVariants.map(variant) => {
+        const localizedPath = `/${variant.locale}/${variant.path}`
+        const localizedPageUrl = new URL(localizedPath, DOMAIN)
+        return {
+          hreflang: variant.locale,
+          href: localizedPageUrl.href
+        }
+      }
+    }
+  })
+  .toArray()
+
+  return getServerSideSitemapLegacy(context, sitemap)
+}
+
+export default function Sitemap() {}
```

Comment:
I think this example is broken, need to double-check. Also, I'd move all of the prop mapping minutiae into a separate local function (`pageToSitemapItem`) so that this one can be just (untested):

```typescript
export async function getServerSideProps(context) {
  const sitemap = client.getPages()
    .filter(page => !page.excludedFromSearch)
    .map(pageToSitemapItem)
    .toArray()

  return getServerSideSitemapLegacy(context, sitemap)
}
```